server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java

```diff
@@ -368,7 +365,7 @@ private PopulationStatus populateOutput() {
         synchronized (this.entriesToWrite) {
             while (!this.terminated && this.entriesToWrite.isEmpty()) {
                 try {
-                    this.entriesToWrite.wait(this.walWaitIntervalMillis);
+                    this.entriesToWrite.wait();
```

Comment:
Will `synchronized` and `wait` forever causes deadlock?

Reply:
As long as it is notified properly, this is not a problem. Anyway, I add a constant here to prevent possible bugs.